src/screens/ConfirmEmail.js

```diff
@@ -0,0 +1,190 @@
+import React, {useEffect, useState} from 'react';
+import {
+  SafeAreaView,
+  View,
+  ScrollView,
+  Text,
+  TouchableOpacity,
+  StyleSheet,
+  Alert,
+} from 'react-native';
+import {Colors} from '../utils/Colors';
+import {NEXT_MESSAGE} from '../utils/Constants';
+import CodeConfirmTxtInput from '../components/CodeConfirmTxtInput';
+
+const ConfirmEmail = ({route}) => {
+  const {usersEmail} = route.params;
+  let codeConfirmation = ['1', '2', '3', '4'];
+  let auxiliarCounter = 0;
+  const [formData, setFormData] = useState({
+    fieldOne: '',
+    fieldTwo: '',
+    fieldThree: '',
+    fieldFour: '',
+  });
+
+  useEffect(() => {
+    setFormData({
+      fieldOne: '',
+      fieldTwo: '',
+      fieldThree: '',
+      fieldFour: '',
+    });
+  }, []);
+
+  const handleOnChange = ({number, id}) => {
+    setFormData({
+      ...formData,
+      [id]: number,
+    });
+  };
+
+  const resetForm = () => {
+    setFormData({
+      fieldOne: '',
+      fieldTwo: '',
+      fieldThree: '',
+      fieldFour: '',
+    });
+  };
+
+  const checkCodeConfirmation = () => {
+    if (
+      formData.fieldOne === '' &&
+      formData.fieldTwo === '' &&
+      formData.fieldThree === '' &&
+      formData.fieldFour === ''
+    ) {
+      Alert.alert('type all the code');
+    } else {
+      let userCodeConfirmation = [
+        formData.fieldOne,
+        formData.fieldTwo,
+        formData.fieldThree,
+        formData.fieldFour,
+      ];
+      for (let i = 0; i < 4; i++) {
+        if (codeConfirmation[i] === userCodeConfirmation[i]) {
+          auxiliarCounter++;
+        }
+      }
+      if (auxiliarCounter === 4) {
+        Alert.alert('Success');
+      } else {
+        Alert.alert('Your code is wrong');
+      }
+    }
+  };
+
+  return (
+    <SafeAreaView style={styles.container}>
+      <ScrollView>
+        <Text style={styles.title}>Confirmación de correo</Text>
+        <Text style={styles.textInstruction}>
+          Por favor ingresa el código enviado a{' '}
+          <Text style={[styles.textInstruction, styles.userEmail]}>
+            {usersEmail}
+          </Text>
+        </Text>
+        <View style={styles.inputsContainer}>
+          <CodeConfirmTxtInput
+            id="fieldOne"
+            value={formData.fieldOne}
+            onChangeText={handleOnChange}
+          />
+          <CodeConfirmTxtInput
+            id="fieldTwo"
+            value={formData.fieldTwo}
+            onChangeText={handleOnChange}
+          />
+          <CodeConfirmTxtInput
+            id="fieldThree"
+            value={formData.fieldThree}
+            onChangeText={handleOnChange}
+          />
+          <CodeConfirmTxtInput
+            id="fieldFour"
+            value={formData.fieldFour}
+            onChangeText={handleOnChange}
+          />
```

Comment:
This code is being repeated, make a function where you iterate and return that JSX.